assignments/array-methods.js

```diff
@@ -54,30 +56,99 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
+// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. 
+//Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
 
+runners.map(function (runner, index) {
+
+    allCaps[index] = runner.first_name.toUpperCase();
+  });
+
+  console.log(allCaps);
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can 
+//choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log 
+//the result
 let largeShirts = [];
-console.log(largeShirts);
+
+largeShirts = runners.filter(function (runner) {
+
+    if (runner.shirt_size === 'L') {
+  
+      runners.filter(function (runner) {
+        return runner.shirt_size === 'L';
+      });
+      return runner;
+    }
+  });
+
+  console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
-// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
+// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the 
+//result
 let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+
+ticketPriceTotal = runners.reduce(function (totalValue, runner) {
+
+    return totalValue + runner.donation;
+  }, 0);
+
+  console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
-// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
+// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given 
+//the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+//The race director needs the emails of all the runners.
+let runnerEmails=[];
+
+runners.forEach(function (runner) {
+    runnerEmails.push(runner.email);
+  });
+
+  console.log(runnerEmails);
 
 // Problem 2
+//Change all last names to lowercase.
+let noCaps= [];
+
+runners.map(function (runner, index) {
+
+    noCaps[index] = runner.last_name.toLowerCase();
+  });
+
```

Comment:
no need for the `noCaps[index] =` here, just `return runner.last_name.toLowerCase()`